torchbenchmark/models/dcgan/__init__.py

```diff
@@ -0,0 +1,380 @@
+        if self.jitshape != input.shape:
+            self.jitshape = input.shape
+            print(input.shape)
+
+            self.jt = torch.jit.trace(self.main, input)
+
+            #self.jt = torch.quantization.fuse_modules(self.jt, [['bn', 'relu']], inplace=False)
+        
+            print(self.jt)
+
+        return self.jt(input)
+
+        if self.jt != None:
+            return self.jt(input)
+
+      return self.main(input)
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.GENERATION
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+
+        self.debug_print = False
+
+        self.jit = jit
+        self.device = device
+        self.root = str(Path(__file__).parent)
+        self.jit = jit
+
+        self.dcgan = DCGAN(self)
+
+        dcgan = self.dcgan
+
+        device = dcgan.device
+        ngpu = dcgan.ngpu
+        nz = dcgan.nz
+        lr = dcgan.lr
+        beta1 = dcgan.beta1
+        num_epochs = dcgan.num_epochs
+
+        # Create the generator
+        self.netG = Generator(dcgan).to(device)
+
+        # Handle multi-gpu if desired
+        if (dcgan.device == 'cuda') and (ngpu > 1):
+            self.netG = nn.DataParallel(netG, list(range(ngpu)))
+
+        # Apply the weights_init function to randomly initialize all weights
+        #  to mean=0, stdev=0.2.
+        self.netG.apply(weights_init)
+
+        if self.debug_print:
+            # Print the model
+            print(self.netG)
+
+        # Create the Discriminator
+        self.netD = Discriminator(dcgan).to(device)
+
+        # Handle multi-gpu if desired
+        if (dcgan.device == 'cuda') and (ngpu > 1):
+            self.netD = nn.DataParallel(netD, list(range(ngpu)))
+
+        # Apply the weights_init function to randomly initialize all weights
+        #  to mean=0, stdev=0.2.
+        self.netD.apply(weights_init)
+        
+        if self.debug_print:
+            # Print the model
+            print(self.netD)
+
+        # Initialize BCELoss function
+        self.criterion = nn.BCELoss()
+
+        # Create batch of latent vectors that we will use to visualize
+        #  the progression of the generator
+        self.fixed_noise = torch.randn(64, nz, 1, 1, device=device)
+
+        # Establish convention for real and fake labels during training
+        self.real_label = 1.
+        self.fake_label = 0.
+
+        # Setup Adam optimizers for both G and D
+        self.optimizerD = optim.Adam(self.netD.parameters(), lr=lr, betas=(beta1, 0.999))
+        self.optimizerG = optim.Adam(self.netG.parameters(), lr=lr, betas=(beta1, 0.999))
+
+        # Train batch size
+        self.train_b_size = 32
+
+        # Inference Choice:
+        #   just run descriminator
+        #   run generator and descriminator
+
+        # inference would just run descriminator so thats what we'll do too.
+        self.inference_just_descriminator = True
+
+        # eval batch size
+        self.eval_b_size = 256
```

Comment:
Maybe it is better to store `eval_b_size` to `class DCGAN`, and assign `self.eval_b_size = dcgan.eval_b_size` here?